### PR TITLE
Feature/better sync

### LINKED ISSUE #16: Add Turntable synchronization
Turntables still need to be synced based on save loading and based on input across users.

### LINKED ISSUE #17: Add better save game synchronization
Currently, the whole save is being sent to the client while the whole save file is not needed. Just send over the values needed for synchronization.

### LINKED ISSUE #19: Add horn synchronization
When using the horn this should be synchronized.

### LINKED ISSUE #23: Add jobs synchronization
Add synchronization to jobs.

### LINKED ISSUE #27: When player disconnects the disconnected player still resides on the server (not for other players)
When the user disconnects from the server, the server object doesn't remove the certain player from the list hence all new players connecting see a ghost player.

### LINKED ISSUE #28: When a train is turned on while the server was up the newcomers don't share this state

### LINKED ISSUE #30: Add user latency (in ms) to name tag
As latency is critical to whether a user stays synced with the server, adding it to the name tag would help diagnose potential issues.

### LINKED ISSUE #35: Cant shut server off still
I just can't shut down the server when i want to turn it off without closing derail valley 
[Player.log](https://github.com/Patrick-van-Halm/dv-multiplayer-mod/files/5934571/Player.log)

### LINKED ISSUE #39: [Custom UI] Change all close buttons to back buttons
As there is now a dedicated button on the main menu, all sub menus should have back buttons rather than direct close buttons to keep a good menu flow

### LINKED ISSUE #41: Spawn player in nearest train station from where server was started
Wherever someone starts a server, spawn the connecting players inside the nearest station from that point

### PR DESCRIPTION
**Pull request by Patrick-van-Halm**
This will overhaul most sync and the majority of the net code.
Will fix the majority of issues

### CODE REVIEW

**Comment by mibmo** [comment]
Can we test these tomorrow?

File: DVMultiplayer/DTO/Player/Location.cs [hunk 1]
```diff
@@ -1,31 +1,30 @@
...
 namespace DVMultiplayer.DTO.Player
 {
     public class Location : IDarkRiftSerializable
     {
         public ushort Id { get; set; }
-        public Vector3 AbsPosition { get; set; }
-        public Quaternion NewRotation { get; set; }
+        public Vector3 Position { get; set; }
+        public Quaternion? Rotation { get; set; }
+        public int RTT { get; set; } = 0;
```
Comment (old-mibmo) [inline DVMultiplayer/DTO/Player/Location.cs]: Add comment describing what RTT is.

File: DVMultiplayer/Patches/PassengerJobs/PassengerJobsGeneratorPatch.cs [hunk 2]
```diff
@@ -1,44 +0,0 @@
-﻿using HarmonyLib;
```
Comment (old-mibmo) [inline DVMultiplayer/Patches/PassengerJobs/PassengerJobsGeneratorPatch.cs]: Just FYI: there's a `git mv` command; you don't have to manually move (rename) then `git rm` and `git add`.

File: DebtPlugin/DebtPlugin.csproj [hunk 3]
```diff
@@ -9,9 +9,10 @@
     <AppDesignerFolder>Properties</AppDesignerFolder>
     <RootNamespace>DebtPlugin</RootNamespace>
     <AssemblyName>DebtPlugin</AssemblyName>
-    <TargetFrameworkVersion>v4.7.2</TargetFrameworkVersion>
+    <TargetFrameworkVersion>v4.8</TargetFrameworkVersion>
     <FileAlignment>512</FileAlignment>
     <Deterministic>true</Deterministic>
+    <TargetFrameworkProfile />
```
Comment (old-mibmo) [inline DebtPlugin/DebtPlugin.csproj:15]: What does this do?

Comment (Patrick-van-Halm) [reply DebtPlugin/DebtPlugin.csproj:15]: Auto generated 🤷

**Pull request merged**